semantic-release/create-config.sh

```diff
@@ -1,5 +1,5 @@
 IS_TRACKED=$(git ls-files package-lock.json || true)
```

Comment:
Out-of-scope, but valuable for catching errors:

```suggestion
set -eu
IS_TRACKED=$(git ls-files package-lock.json || true)
```
`-e` ~= throw on error and `-u` ~= error on unset variable (https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html).

Reply:
Weirdly doing this in my test action didn't cause anything to fail:
https://github.com/dlockhart/semantic-release-action-test/runs/2353894542?check_suite_focus=true

Reply:
I'm kinda really confused now. I ran this locally using Git Bash without issues (note the lack of `"`s around `${IS_TRACKED}`):

`set-e-iff.sh`: 
```bash
#!/bin/bash

set -eu

IS_TRACKED=a
if [ $IS_TRACKED != "" ]; then
  ASSETS="[\"package.json\", \"package-lock.json\"]"
else
  ASSETS="[\"package.json\"]"
fi

echo ${ASSETS}
```

```bash
$ ./set-e-iff.sh
["package.json", "package-lock.json"]
```

The action says to use bash... https://github.com/BrightspaceUI/actions/blob/abfe9e5cb06ff48b74cd1bd6490a22b4f7630839/semantic-release/action.yml#L32-L42

Maybe it's related to bash version or OS?

```bash
$ bash --version
GNU bash, version 4.4.23(1)-release (x86_64-pc-msys)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

Reply:
This will fail, so I'm guessing errors inside `if`s don't count for `set -e`?

```bash
#!/bin/bash

set -eu
grep a b
```

```bash
$ ./set-e-iff.sh
grep: b: No such file or directory
```

Reply:
I'm comfortable merging as-is or with `set -eu` added. I think there's value in adding the `"`s.

Reply:
Yeah I definitely feel like this was working at one point in the not-so-distant past. Will merge as-is.